ui-tests/src/test/java/io/jenkins/plugins/analysis/warnings/WarningsNextGenerationPluginTest.java

```diff
@@ -0,0 +1,611 @@
+
+        AnalysisResult resultPage = new AnalysisResult(build, "checkstyle");
+        resultPage.open();
+
+        IssuesTable issuesTable = resultPage.openIssuesTable();
+        assertThat(issuesTable).hasSize(1);
+    }
+
+    private void reconfigureJobWithResource(final FreeStyleJob job, final String fileName) {
+        job.configure(() -> job.copyResource(WARNINGS_PLUGIN_PREFIX + fileName));
+    }
+
+    /**
+     * Creates and builds a maven job and verifies that all warnings are shown in the summary and details views.
+     */
+    @Test
+    @WithPlugins({"maven-plugin", "analysis-model-api@7.0.4"})
+    public void should_show_maven_warnings_in_maven_project() {
+        MavenModuleSet job = createMavenProject();
+        copyResourceFilesToWorkspace(job, SOURCE_VIEW_FOLDER + "pom.xml");
+
+        IssuesRecorder recorder = job.addPublisher(IssuesRecorder.class);
+        recorder.setToolWithPattern("Maven", "");
+        recorder.setEnabledForFailure(true);
+
+        job.save();
+
+        Build build = buildFailingJob(job);
+        build.open();
+
+        AnalysisSummary summary = new AnalysisSummary(build, MAVEN_ID);
+        assertThat(summary).isDisplayed()
+                .hasTitleText("Maven: 2 warnings")
+                .hasNewSize(0)
+                .hasFixedSize(0)
+                .hasReferenceBuild(0);
+
+        AnalysisResult mavenDetails = summary.openOverallResult();
+        assertThat(mavenDetails).hasActiveTab(Tab.MODULES)
+                .hasTotal(2)
+                .hasOnlyAvailableTabs(Tab.MODULES, Tab.TYPES, Tab.ISSUES);
+
+        IssuesTable issuesTable = mavenDetails.openIssuesTable();
+
+        DefaultWarningsTableRow firstRow = issuesTable.getRowAs(0, DefaultWarningsTableRow.class);
+        ConsoleLogView sourceView = firstRow.openConsoleLog();
+        assertThat(sourceView).hasTitle("Console Details")
+                .hasHighlightedText("[WARNING]\n"
+                        + "[WARNING] Some problems were encountered while building the effective model for edu.hm.hafner.irrelevant.groupId:random-artifactId:jar:1.0\n"
+                        + "[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-compiler-plugin is missing. @ line 13, column 15\n"
+                        + "[WARNING]\n"
+                        + "[WARNING] It is highly recommended to fix these problems because they threaten the stability of your build.\n"
+                        + "[WARNING]\n"
+                        + "[WARNING] For this reason, future Maven versions might no longer support building such malformed projects.\n"
+                        + "[WARNING]");
+    }
+
+    /**
+     * Verifies that warnings can be parsed on a agent as well.
+     */
+    @Test
+    @WithDocker
+    @WithPlugins("ssh-slaves")
+    @WithCredentials(credentialType = WithCredentials.SSH_USERNAME_PRIVATE_KEY, values = {CREDENTIALS_ID, CREDENTIALS_KEY})
+    public void should_parse_warnings_on_agent() {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The JUnit 4 test method name 'should_parse_warnings_on_agent' doesn't match '[a-z][a-zA-Z0-9]*'](https://app.codacy.com/manual/uhafner/warnings-ng-plugin/pullRequest?prid=5070362)